pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -362,6 +371,12 @@ func (r *ReconcileClusterSync) Reconcile(request reconcile.Request) (reconcile.R
 
 	setFailedCondition(clusterSync)
 
+	// Set clusterSync.Status.FirstSyncSetsSuccessTime
+	syncStatuses := append(syncStatusesForSyncSets, syncStatusesForSelectorSyncSets...)
+	if len(syncStatuses) > 0 && clusterSync.Status.FirstSuccessTime == nil && !selectorSyncSetsNeedRequeue && !syncSetsNeedRequeue {
```

Comment:
I don't think that the checks for the `***NeedRequeue` variables are necessary. What we really want to check is that all of the sync statuses have results of success, which is kind of what is being done in `setFirstSuccessTime`.

Reply:
Removed `***NeedRequeue` checks and now just rely on `setFirstSuccessTime`.